detox/android/detox/src/full/java/com/wix/detox/espresso/MultiViewActionWithResult.kt

```diff
@@ -0,0 +1,45 @@
+package com.wix.detox.espresso
+
+import android.view.View
+import androidx.test.espresso.Espresso
+import org.hamcrest.Matcher
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+abstract class MultiViewActionWithResult<R> : ViewActionWithResult<R> {
+    fun performOnView(viewMatcher: Matcher<View>): Any {
```

Comment:
The name is misleading because the action is performed over _multiple_ views (potentially), based on a matcher.